apps/pie-storybook/stories/pie-switch.stories.ts

```diff
@@ -82,23 +110,89 @@ const changeAction = (event: Event) => action('change')({
 const Template : TemplateFunction<SwitchProps> = (props) => {
     const {
         aria,
-        isChecked,
+        checked,
         isDisabled,
         label,
         labelPlacement,
+        name,
+        value,
+        required,
     } = props;
 
     return html`
         <pie-switch
+            id="pie-switch"
+            name="${name || nothing}"
+            value="${value || nothing}"
             label="${label || nothing}"
             labelPlacement="${label && labelPlacement ? labelPlacement : nothing}"
             .aria="${aria}"
-            ?isChecked="${isChecked}"
+            ?required="${required}"
+            ?checked="${checked}"
             ?isDisabled="${isDisabled}"
             @change="${changeAction}">
         </pie-switch>`;
 };
 
+const FormTemplate: TemplateFunction<SwitchProps> = (props: SwitchProps) => html`
+    <p id="formLog" style="display: none; font-size: 2rem; color: var(--dt-color-support-positive);"></p>
+    <h2>Fake form</h2>
+    <form id="testForm">
+    <p>Required fields are followed by <strong><span aria-label="required">*</span></strong>.</p>
+
+    <section>
+    ${Template({
+    ...props,
+})}
+    </section>
+    <button type="submit">Submit</button>
+    </form>
+    <script>
+        // Display a success message to the user when they submit the form
+        const form = document.querySelector('#testForm');
+        const formLog = document.querySelector('#formLog');
+
+        const el = document.querySelector('pie-switch');
+        const s = document.querySelector('pie-switch');
+        setTimeout(() => {
+        console.log(s);
+        // console.log('checking the report v value');
+        // s.setCustomValidity('wang it in, son');
+        // console.log(c.reportValidity());
+        // console.log('calling check validity');
+        }, 0);
+
+
+
+        form.addEventListener('submit', (e) => {
+            e.preventDefault();
+
+            el.checkValidity();
+
+            formLog.innerHTML = 'Form submitted!';
+            formLog.style.display = 'block';
+
+            // log out all form input values
+            const formData = new FormData(form);
+            console.log('All form elements:', form.elements);
+            console.log('All form element data keys and values submitted:');
+
+            for (const entry of formData.entries()) {
+                console.table(entry);
+            }
+
+            // Reset the success message after roughly 8 seconds
+            setTimeout(() => {
+                formLog.innerHTML = '';
+                formLog.style.display = 'none';
+            }, 8000);
```

Comment:
Not important at all but I noticed if you submit the form twice within those 8 seconds the reset from the first click will apply to the second

Reply:
this is debug code I didn't mean to commit!

Reply:
regarding the double submit, I don't think it matters for the sake of this story. It's just minimal code to add some extra visibility of integration